File: Liquid.Core/Entities/Customer.cs
﻿using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Liquid.Core.Entities
{
    public class Customer : BaseEntity
    {
        public Customer()
        {

        }

        [Key, DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int CustomerId { get; set; }
        public Guid CustomerGuId { get; set; }
        public string Name { get; set; }
        public string Phone { get; set; }
        public string AddressLine1 { get; set; }
        public string? AddressLine2 { get; set; }
        public string City { get; set; }
        public string? State { get; set; }
        public string PostalCode { get; set; }
    }
}

File: Liquid.UI/src/app/shared/models/customer.ts
import { IBaseEntity } from "./base.entity";
export interface ICustomer extends IBaseEntity {
    customerId: number;
    customerGuId: string;
    name: string;
    phone: string;
    addressLine1: string;
    addressLine2?: string;
    city: string;
    state?: string;
    postalCode: string;
}